### PR TITLE
Add the `preset` parameter

### PR DESCRIPTION
**Pull request by robertkozin**
Add support for the `preset` param.

The preset param is: The ID of a custom playground preset. You can create presets in the [playground](https://os.cohere.ai/playground). If you use a preset, all other parameters become optional, and any included parameters will override the preset's parameters.

### CODE REVIEW

File: classify_test.go
Comment (lfayoux) [inline classify_test.go:97]: Will this pass once the preset work is merged?

Comment (robertkozin) [reply classify_test.go:97]: Yeah, I'm waiting for the API side to be merged, then I will update PR with real preset and make sure it passes

File: generate.go
Comment (lfayoux) [inline generate.go:31]: Does it mater if `Prompt` isn't `omitempty`? I image the preset will take precedent over all other params

Comment (robertkozin) [reply generate.go:31]: Oh good catch, thanks. It does matter, the params made inside of the request will override the params of the preset.

File: generate.go
Comment (lfayoux) [inline generate.go:33]: Should this also be `omitempty`?

**Review by lfayoux — APPROVED** [review]

**Pull request merged**
